xarray/core/dataset.py

```diff
@@ -4071,8 +4072,55 @@ def to_dataframe(self):
         """
         return self._to_dataframe(self.dims)
 
+    def _set_sparse_data_from_dataframe(self, dataframe, dims, shape):
```

Comment:
Albeit boring, I find that putting type annotations in the internal functions helps a lot in weeding out subtle bugs.

Reply:
done